--- ./clean.sh	2019-09-23 16:53:11.748086779 +0300
+++ ./clean.sh	2019-09-23 17:04:11.842249468 +0300
@@ -6,25 +6,31 @@
 localRegistryStarted=`runningDockerContainer docker-registry`
 
 if [ -z "$localRegistryStarted" ] ; then
-    docker rm -f $(docker ps -aq)
+     docker ps -a | grep -v jenkins | tail -n+2 | cut -d ' ' -f 1 | xargs --no-run-if-empty docker rm -f
 else
     echo "localRegistryStarted=$localRegistryStarted"
     killContainers=`docker ps -aq | sed -e "s/${localRegistryStarted}/ /"`
     echo "killContainers=$killContainers"
-    docker rm -f ${killContainers}
+    docker rm -f ${killContainers}; 
+    
 fi
 
 #TODO [ "${DOCKER_MACHINE_NAME}" == "orderer" ]  && EXECUTE_BY_ORDERER=1 runCLIWithComposerOverrides down || runCLIWithComposerOverrides down
 
 docker volume prune -f
-docker rmi -f $(docker images -q -f "reference=dev-*")
+docker images -q -f "reference=dev-*" | xargs --no-run-if-empty docker rmi -f 
 
 if [ -z "$DOCKER_HOST" ] ; then
     docker-compose -f docker-compose-clean.yaml run --rm cli.clean rm -rf crypto-config/*
-    [ "$all" == "all" ] && docker-compose -f docker-compose-clean.yaml run --rm cli.clean rm -rf data/*
+       if [ "$all" == "all" ] ; then
+	    docker-compose -f docker-compose-clean.yaml run --rm cli.clean rm -rf data/*
+	fi
 else
     docker-machine ssh ${DOCKER_MACHINE_NAME} sudo rm -rf crypto-config
-    [ "$all" == "all" ] && docker-machine ssh ${DOCKER_MACHINE_NAME} sudo rm -rf data/
+
+    if [ "$all" == "all" ] ; then
+	docker-machine ssh ${DOCKER_MACHINE_NAME} sudo rm -rf data/
+    fi
 fi
 
 #docker rmi -f $(docker images -q -f "reference=olegabu/fabric-starter-client")
 